File: client/src/app/services/admin.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const ADMIN_API = 'http://localhost:8080/api/admin/';

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  constructor(private httpClient: HttpClient) {
  }

  updateOrderPrice(orderId: number): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'update-order-price/' + orderId);
  }

  acceptOrder(orderId: number): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'accept-order/' + orderId);
  }

  getAllDrivers(): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-all-drivers');
  }

  getAllUsers(): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-all-users');
  }

  getAllOrders(): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-all-orders');
  }

  getAllOrdersByDriver(driver: string): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-all-orders-by-driver/' + driver);
  }

  getAllOrdersByUser(username: string): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-all-orders-by-user/' + username);
  }

  getOrderItems(orderId: number): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-order-items/' + orderId);
  }

  hireDriver(driver: any): Observable<any> {
    return this.httpClient.post(ADMIN_API + 'hire-driver', {
      username: driver.username,
      password: driver.password,
    });
  }

  rejectOrder(orderId: number): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'reject-order/' + orderId);
  }

  getAllNotifications(): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-all-notifications');
  }

  removeNotification(notificationId: number): Observable<any> {
    return this.httpClient.delete(ADMIN_API + 'remove-notification/' + notificationId);
  }

  confirmDriverRating(driverUsername: string): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'confirm-driver-rating/' + driverUsername);
  }

  removeDriverRating(ratingId: number): Observable<any> {
    return this.httpClient.delete(ADMIN_API + 'remove-driver-rating/' + ratingId);
  }

  getDriverRating(driverUsername: string): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-driver-rating/' + driverUsername);
  }

  getAllUsersPagination(page: number): Observable<any> {
    return this.httpClient.get(ADMIN_API + 'get-all-users-pagination/' + page);
  }
}

File: server/src/main/java/by/belstu/deliverty/services/AdminService.java
package by.belstu.deliverty.services;

import by.belstu.deliverty.dto.*;
import by.belstu.deliverty.payload.SignupRequest;
import by.belstu.deliverty.repository.AdminRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class AdminService {
    private final AdminRepository adminRepository;

    @Autowired
    public AdminService(AdminRepository adminRepository) {
        this.adminRepository = adminRepository;
    }

    public void updateOrderPrice(Long orderId) {
        adminRepository.updateOrderPrice(orderId);
    }

    public void acceptOrder(Long orderId) {
        adminRepository.acceptOrder(orderId);
    }

    public void hireDriver(SignupRequest signupRequest) {
        adminRepository.hireDriver(signupRequest.getUsername(), signupRequest.getPassword());
    }

    public List<DriverDTO> getAllDrivers() {
        return adminRepository.getAllDrivers();
    }

    public List<UserDTO> getAllUsers() {
        return adminRepository.getAllUsers();
    }

    public List<AdminOrderDTO> getAllOrders() {
        return adminRepository.getAllOrders();
    }

    public List<AdminOrderDTO> getAllOrdersByDriver(String driverUsername) {
        return adminRepository.getAllOrdersByDriver(driverUsername);
    }

    public List<AdminOrderDTO> getAllOrdersByUser(String username) {
        return adminRepository.getAllOrdersByUser(username);
    }

    public List<OrderItemsDTO> getAllOrderItems(Long orderId) {
        return adminRepository.getAllOrderItems(orderId);
    }

    public void rejectOrder(Long orderId) {
        adminRepository.rejectOrder(orderId);
    }

    public List<NotificationsDTO> getAdminNotifications() {
        return adminRepository.getAdminNotifications();
    }

    public void removeNotification(Long notificationId) {
        adminRepository.removeNotification(notificationId);
    }

    public void confirmDriverRating(String driverUsername) {
        adminRepository.confirmDriverRating(driverUsername);
    }

    public void removeDriverRating(Long ratingId) {
        adminRepository.removeDriverRating(ratingId);
    }

    public List<RatingDTO> getDriverRating(String driverUsername) {
        return adminRepository.getDriverRating(driverUsername);
    }

    public List<UserDTO> getAllUsersPagination(Long page) {
        return adminRepository.getAllUsersPagination(page);
    }
}
